src/ExpressionEvaluator/Core/Source/ResultProvider/Helpers/TypeHelpers.cs

```diff
@@ -439,9 +434,21 @@ internal static DkmCustomUIVisualizerInfo[] GetDebuggerCustomUIVisualizerInfo(th
             return result;
         }
 
-        private static string EvaluateDebuggerDisplayString(DkmClrValue value, string str)
+        internal static void EvaluateDebuggerDisplayStringAndContinue(this DkmClrValue value, DkmWorkList workList, DkmInspectionContext inspectionContext, DkmClrType targetType, string str, DkmCompletionRoutine<DkmEvaluateDebuggerDisplayStringAsyncResult> completionRoutine)
+        {
+            if (str == null)
+            {
+                completionRoutine(default(DkmEvaluateDebuggerDisplayStringAsyncResult));
+            }
+            else
+            {
+                value.EvaluateDebuggerDisplayString(workList, inspectionContext, targetType, str, completionRoutine);
+            }
+        }
+
+        internal static string GetResult(this DkmEvaluateDebuggerDisplayStringAsyncResult asyncResult)
         {
-            return str == null ? null : value.EvaluateDebuggerDisplayString(str);
+            return !asyncResult.Equals(default(DkmEvaluateDebuggerDisplayStringAsyncResult)) ? asyncResult.Result : null;
```

Comment:
So, the problem is that I need to avoid calling EvaluateDebuggerDisplayString (the real one) with a null FormatString.  The way I'm indicating that the FormatString provided by the Attribute was null is by using default(DkmEvaluateDebuggerDisplayStringAsyncResult), because I can't construct one of these using null as the argument.  It would seem to complicate the logic of GetResultAndContinue significantly if I have to check the Attribute values for null and avoid calling EvaluateDebuggerDisplayString there (because of the continuation passing).  Thoughts?